sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/schemas/namespaces/NamespacesResource.java

```diff
@@ -0,0 +1,404 @@
+                          properties =
+                              @SchemaProperty(
+                                  name = "data",
+                                  type = SchemaType.ARRAY,
+                                  implementation = NamespaceDto.class,
+                                  uniqueItems = true,
+                                  minItems = 0)))
+            }),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @GET
+  public Uni<RestResponse<Object>> getAllNamespaces(@QueryParam("raw") boolean raw) {
+
+    // get all
+    return namespaceManager
+        .getNamespaces()
+
+        // then map and collect
+        .map(this::toNamespaceDto)
+        .collect()
+        .asList()
+
+        // map to wrapper if needed
+        .map(
+            results -> {
+              if (raw) {
+                return results;
+              } else {
+                return new SimpleResponseWrapper<>(results);
+              }
+            })
+        .map(RestResponse::ok);
+  }
+
+  @Operation(description = "Retrieve a single namespace specification.")
+  @Parameters(
+      value = {
+        @Parameter(
+            name = "namespace",
+            ref = OpenApiConstants.Parameters.NAMESPACE,
+            description = "The namespace name."),
+        @Parameter(name = "raw", ref = OpenApiConstants.Parameters.RAW),
+      })
+  @APIResponses(
+      value = {
+        @APIResponse(
+            responseCode = "200",
+            description =
+                "Call successful. Note that in case of unwrapping (`raw=true`), the response contains only the contents fof the `data` property.",
+            content = {
+              @Content(
+                  schema =
+                      @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                          implementation = SimpleResponseWrapper.class,
+                          properties =
+                              @SchemaProperty(name = "data", implementation = NamespaceDto.class)))
+            }),
+        @APIResponse(
+            responseCode = "404",
+            description = "Not found.",
+            content =
+                @Content(
+                    examples = {
+                      @ExampleObject(ref = OpenApiConstants.Examples.NAMESPACE_DOES_NOT_EXIST)
+                    },
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @GET
+  @Path("/{namespace:\\w+}")
```

Comment:
I'm assuming this `\w` refers to the Regex class [A-Za-z0-9]+, which would not match certain characters that are technically allowed to be in keyspaces: [see here](https://docs.datastax.com/en/cql-oss/3.x/cql/cql_reference/ref-lexical-valid-chars.html#:~:text=characters%20in%20names-,Keyspace%20and%20table%20names%20can%20only%20contain%20alpha%2Dnumber%20characters,alpha%2Dnumeric%20characters%20and%20underscores.)

For instance, a keyspace like `eric-keyspace` is allowed as long as the value is enclosed in double-quotes (even though hyphen is a character that would not be matched by `\w`)

Reply:
Oh if that's the case, then never mind! Yeah you're right, that restriction has existed since the beginning.

But then `\w` should suffice and you shouldn't need `\w+` ...right?

Reply:
underscore char is missing